### DISCUSSION TITLE
Is it possible to automatically generate a C++ class from json and store it as .cpp and .h files?

### DISCUSSION THREAD

**Discussion by Silver-will** [nlohmann/json#4300] [Q&A]
I was wondering if it was possible to auto generate and map data structures, for several json files i have all with differing keys to avoid having to manually define each structure for the 50+ separate json documents i have.
Something like what https://github.com/beached/json_to_cpp does.

**Comment by nlohmann** [comment]
nlohmann/json itself has no code-generation tool built in; it only provides the NLOHMANN_DEFINE_TYPE_INTRUSIVE/NON_INTRUSIVE macros (include/nlohmann/detail/macro_scope.hpp) to cut down boilerplate once you've already written the struct. For actually generating C++ struct definitions and (de)serialization code from sample JSON files, you'd need an external tool such as quicktype (which has an nlohmann/json output mode) or json_to_cpp, as referenced in the question. There's no first-party equivalent in this repository.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
